Etiquette/CODE_REVIEW.md

```diff
@@ -0,0 +1,108 @@
+* Avoid selective ownership of code ("mine", "not mine", "yours"). 
+* Use "we" instead of "you", unless asking for opinion of specific individual ("What do you think ...?", but "Here we are doing ... What if we ...?",)
+* Avoid using terms that could be seen as referring to personal traits. ("dumb",
+  "stupid", "insane"). Assume everyone is intelligent and well-meaning.
+* Be aware of negative bias with online communication. (If content is neutral, we assume the tone is negative.) Can you use positive language as opposed to neutral?
+* When using emojis be aware that their meaning can be different for different individuals - winking smile may not be always innosent
+* Same with any other media format you decide to use
+* Do not expect your comments to be answered immediately. If you need an answer fast - talk in person (same applies to the Slack messages)
+* It's fine to have disagreements and its hard to please everyone.
+
+
+Creating Pull Request
+--------
+
+* Carefully fill in pull request template:
+	* title of the PR should follow format `[XXX-123] Description of the change` starting with a ticket number
+	* explain the context and motivation, don’t assume familiarity with the history. Highlight how the goal was achived without going too much into details - people tend not to read long descriptions and if the long explanation is needed probably just pull request will be not enough (the change may need to go through proposal process). 
+	* don't strip out parts of the template unless it is absolutely unrelevant to the changes (i.e. change in the build script does not need any screenshots as it does not touch UI)
+	* don't forget labels (it's better to set them after PR is created as github sometimes fails with 405 status code and we don't know why)
+* explicitly request review from or mention team mates you specifically want to involve in the discussion
+* Self-review your changes, adding comments where you think you can get questions (it may worth adding these as code or documentation comments). You can use github suggestions and apply them right away
+* Keep pull requests small, ideally it shouldn't be more than **800 additions/deletions**
+* Create a draft PR for work that is in progress, but only do that if you seek for opinions on your work
+* Mark PR with "Blocked" label if it has any dependency that should stop it from being merged (it's fine for pull request be both "blocked" and "ready for review", but it shouldn't be "ready for review" and "wip")
+* Give your team mates a reasonable time to review your changes, if the change is critical use "Top priority" label and post it in the team channel
+* Do not leave stale pull requests, seek for help/actions or close them (make sure you don't delete your local branch if you may need it later)
+* It's our common effort to push product forward and we all do equally valuable work. That in mind when creating PR try to review some PRs, avoid having more than X opened PRs
```

Comment:
we should indeed. @NSMyself did you have a chance to look at our stats? I'd say we could maybe put 3 here as we concluded 4 is probably too big considering the size of the team and 2 feels too restrictive as it does not take into account the complexity of PRs (there might be more really trivial PRs).

P.S. I think that's not something that should be strictly enforced by our tools btw.

Reply:
Nope, sorry, I've been focusing on HM. No time right now

Reply:
We might also reevaluate that value once we enforce smaller PRs and encourage splitting features into iterative PRs – like @Ben-Henshall suggested – since that might mean smaller / more trivial PRs... but more PRs per team member too?